File: 217.存在重复元素.js
/*
 * @lc app=leetcode.cn id=217 lang=javascript
 *
 * [217] 存在重复元素
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  let map = new Map() // <元素，出现次数>
  for (element of nums) {
    if (!map.has(element)) {
      map.set(element, 1)
    } else {
      let counts = map.get(element)
      if (counts + 1 > 1) {
        return true
      } else {
        map.set(element, counts + 1)
      }
    }
  }
  return false
};
// @lc code=end

File: 217.存在重复元素.go
/*
 * @lc app=leetcode.cn id=217 lang=golang
 *
 * [217] 存在重复元素
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
	record := make(map[int]bool, len(nums))
	for _, v := range nums {
		if _, ok := record[v]; ok {
			return true
		}
		record[v] = true
	}
	return false
}

// @lc code=end

